### PR TITLE
Fix fetching sourcemap in genymotion. Fixes #5338

### LINKED ISSUE #5338: [HMR] Ensure Hot Loading is working on Android
We've done quite many changes to HMR since @satya164 added the Hot Loading option to the dev menu. This task is to make sure Hot Loading works on a brand new project. Also, we need to gate the feature for now, we don't want to release up until we're sure it's stable.

### PR DESCRIPTION
**Pull request by satya164**
Source maps are broken on Genymotion right now as they aren't being loaded from the correct URL. refer - https://github.com/facebook/react-native/issues/5338#issuecomment-188232402

**Test plan**

Build and install UIExplorer from master branch in genymotion and enable hot reload. When you change a file and save it, you'll see a Yellow box due to source map fetching failed, as per the referenced comment.

Doing the same for this branch doesn't produce any yellow boxes.

### CODE REVIEW

File: Libraries/Utilities/HMRClient.js [hunk 1]
```diff
@@ -73,6 +73,7 @@ Error: ${e.message}`
     };
     activeWS.onmessage = ({data}) => {
       // Moving to top gives errors due to NativeModules not being initialized
+      const { AndroidConstants } = require('NativeModules');
       const HMRLoadingView = require('HMRLoadingView');
```
Comment (martinbigio) [inline Libraries/Utilities/HMRClient.js]: `AndroidConstants` doesn't exist on iOS :)

**Comment by mkonicek** [comment]
Thanks for jumping on this @satya164! Please check out the comment from Martín.

We should probably cherry-pick this to 0.22 and 0.23.

File: Libraries/Utilities/HMRClient.js (hunk 1 above)
Comment (martinbigio) [inline Libraries/Utilities/HMRClient.js]: you can use the `host` and `port` param `HMRClient` receives on the constructor for iOS

**Comment by martinbigio** [comment]
@facebook-github-bot shipit

File: ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/AndroidInfoModule.java [hunk 2]
```diff
@@ -9,14 +9,15 @@
 
 package com.facebook.react.modules.systeminfo;
 
-import javax.annotation.Nullable;
+import android.os.Build;
+
+import com.facebook.react.bridge.BaseJavaModule;
+import com.facebook.react.devsupport.DevServerHelper;
```
Comment (bestander) [inline ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/AndroidInfoModule.java]: Have a look at the CI builds, BUCK file of systeminfo needs a dependency on devsupport

Comment (satya164) [inline ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/AndroidInfoModule.java]: @bestandr Thanks. Trying to fix. How can I verify that buck build passes locally? Sorry for the noob question.

Comment (bestander) [inline ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/AndroidInfoModule.java]: Same as CircleCI.
The quickest way that catches 99%
```
buck fetch ReactAndroid/src/main/java/com/facebook/react
buck build ReactAndroid/src/main/java/com/facebook/react
buck fetch ReactAndroid/src/main/java/com/facebook/react/shell
buck build ReactAndroid/src/main/java/com/facebook/react/shell
buck fetch ReactAndroid/src/test/... 
buck test ReactAndroid/src/test/... 
```

Comment (satya164) [inline ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/AndroidInfoModule.java]: @bestander Thank you.

**Comment by bestander** [comment]
well done, looks green now

**Comment by bestander** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**